lte/gateway/c/core/oai/lib/s6a_proxy/proto_msg_to_itti_msg.cpp

```diff
@@ -134,6 +134,27 @@ void convert_proto_msg_to_itti_s6a_update_location_ans(
     itti_msg->subscription_data.access_mode = NAM_ONLY_PACKET;
   }
 
+  // Regional subscription zone codes
+  itti_msg->subscription_data.num_zcs =
+      ((msg.regional_subscription_zone_code_size() > MAX_REGIONAL_SUB) ?
+           MAX_REGIONAL_SUB :
+           (msg.regional_subscription_zone_code_size()));
+  uint8_t itr = 0;
+  while (itr < itti_msg->subscription_data.num_zcs) {
+    auto regional_subscription_zone_code =
+        msg.regional_subscription_zone_code(itr);
+    // Copy only if the zonecode is <=2 octets
+    if (regional_subscription_zone_code.length() <= ZONE_CODE_LEN) {
+      memcpy(
+          itti_msg->subscription_data.reg_sub[itr].zone_code,
+          regional_subscription_zone_code.c_str(), ZONE_CODE_LEN);
+    } else {
+      std::cout << "[ERROR] Invalid zonecode length received, ignoring"
+                << regional_subscription_zone_code.length() << std::endl;
+    }
+    ++itr;
+  }
```

Comment:
you are not considering the skipped zone codes in forming the itti_msg->subscription_data